main.js

```diff
@@ -1,63 +1,101 @@
-
-const submitButton = document.getElementById('submit');
-submitButton.addEventListener('click', event => {
-  sortSelect.value = 'relevance';
-  submitNewSearch(event);
-});
-
-const toggleFiltersButton = document.getElementById('toggle-filters');
-toggleFiltersButton.addEventListener('click', handleToggleFiltersClick);
-
-const previousPageButton = document.getElementById('prev-page-btn');
-previousPageButton.addEventListener('click', handlePaginationClick);
-
-const nextPageButton = document.getElementById('next-page-btn');
-nextPageButton.addEventListener('click', handlePaginationClick);
-
 let articles;
-let totalHits;
 let resultsPage;
 
-function displayArticles() {
-  const searchResults = document.querySelector('.search-results');
+const sortSelect = document.getElementById('sort-by-select');
+const submitButton = document.getElementById('submit');
+const filtersButton = document.getElementById('filters-button');
+const previousPageButton = document.getElementById('previous-page-button');
+const nextPageButton = document.getElementById('next-page-button');
+
+bindEvents();
+
+function bindEvents() {
+	sortSelect.addEventListener('change', () => {
+		resultsPage = 0;
+		fetchArticles().then(() => {
+			displaySearchResults();
+		});
+	});
+
+	submitButton.addEventListener('click', event => {
+		event.preventDefault();
+		resultsPage = 0;
+		sortSelect.value = 'relevance';
+		fetchArticles().then(() => {
+			displaySearchResults();
+		});
+	});
+	
+	filtersButton.addEventListener('click', event => {
+		event.preventDefault();
+		toggleFilterMenuVisibility();
+	});
+
+	previousPageButton.addEventListener('click', () => {
+		resultsPage--;
+		fetchArticles().then(() => {
+			displaySearchResults();
+		});
+		scroll(0, 0);
+	});
+
+	nextPageButton.addEventListener('click', () => {
+		resultsPage++;
+		fetchArticles().then(() => {
+			displaySearchResults();
+		});
+		scroll(0, 0);
+	});
+}
+
+function displaySearchResults() {
+	const searchResultsDiv = document.getElementById('search-results-container');
+  const articlesDiv = document.getElementById('articles-container');
+	
+	searchResultsDiv.style.display = '';
   
-  while (searchResults.firstChild) {
-    searchResults.removeChild(searchResults.firstChild);
+  while (articlesDiv.firstChild) {
+    articlesDiv.removeChild(articlesDiv.firstChild);
   }
+	
+  const totalHits = articles.response.meta.hits;
+  const totalHitsPara = document.getElementById('total-hits-msg');
+  totalHitsPara.textContent = `Your query returned ${totalHits} hits.`;
+  totalHitsPara.style.display = 'block';
 
-  const articlesOnPage = articles.response.docs;
+  if (totalHits > 0) {
+    const currentPageArticles = articles.response.docs;
 
-  if (articlesOnPage.length > 0) {
-    articlesOnPage.forEach(article => {
-      const articleDiv = document.createElement('div');
-      articleDiv.setAttribute('class', 'article');
+    currentPageArticles.forEach(article => {
```

Comment:
assembling an item like this is not very readable because we can not see the html structure straight away. mrefactoring idea: https://www.w3schools.com/TagS/tag_template.asp

by the way, this particular use case (generate component markup) is where React shines.